CMakeLists.txt

```
@@ -23,6 +23,7 @@ option(SOCI_STATIC "Enable build of static libraries" ON)
 option(SOCI_TESTS "Enable build of collection of SOCI tests" ON)
 option(SOCI_ASAN "Enable address sanitizer on GCC v4.8+/Clang v 3.1+" OFF)
 option(SOCI_LTO "Enable link time optimization" OFF)
+option(SOCI_VISIBILITY "Enable symbol trimming via C++ visibility: Preset visibility to hidden" on)
```

Sorry for nitpicking but I think correct terminology is important: visibility is a property of ELF binaries, not C++ language, so I'd describe it like this:
```suggestion
option(SOCI_VISIBILITY "Enable hiding private symbol using ELF visibility if supported by the platform" on)
```